### PR TITLE
feat: hale»studio 5.1.0

### PR DESCRIPTION
**Pull request by emanuelaepure10**
Update CHANGELOG, add description of the new release to <release-tag>.xhtml page and update infocenter container.

ING-3990

### CODE REVIEW

**Comment by emanuelaepure10** [comment]
Please review the release notes here
https://github.com/halestudio/hale/releases/edit/untagged-6937b5f6747158127f19

and tell me if you believe that something else should be added to the infocenter container.

Thank you

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,42 @@
 All notable changes to this project will be documented in this file.
 See the [change log guidelines](http://keepachangelog.com/) for information on how to structure the file.
 
-## [Unreleased]
+## [5.1.0]
```
Comment (stempler) [inline CHANGELOG.md:5]: The link here would be `https://github.com/halestudio/hale/compare/5.0.1...5.1.0`

The `Unreleased` link also needs to be updated.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,42 @@
...
+- Fix Excel reader not adding instances for empty rows
+- Fix project settings, warnings, include JSON UI extension definition in the build
+- Fix Shapefile writer to recognize geometries for feature types
+- Fix exporting transformed data leading to a shift of attribute values in Shapefile
+- Fix TopoJSON encoding
+- Fix endless loop when skipping instances in paginated WFS request
+- Fix exporting HSD schemas for schemas with no namespace
+
 
 ## [5.0.1]
```
Comment (stempler) [inline CHANGELOG.md:42]: The link here would be `https://github.com/halestudio/hale/compare/5.0.0...5.0.1`

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,7 +2,42 @@
...
+### Added
+- Support for reading multiple sheets from an Excel file simultaneously
+- Enable exporting of multiple feature types to Excel
+- Allow loading Excel with blank rows and sheets for schema and source data
+- Add predefined and custom formatters for Date cells in Excel import
+- Enable exporting feature types with no data in Excel
+- Align Excel with CSV export wizard
+- Prefer properties with short paths as GeoJSON geometry property
+- Add wizard page to select the read mode for GeoJSON/JSON
+- Extend JSON reader functionality
```
Comment (stempler) [inline CHANGELOG.md:16]: There was no Json reader before, so that should be mentioned as new feature.
Small pieces of the feature or fixes that were done to it before release should not be mentioned.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,7 +2,42 @@
...
+- Support loading schema from JSON data
+- Configure decimal places in JSON
```
Comment (stempler) [inline CHANGELOG.md:18]: From the description it is hard to determine what the change is.
Is this for export and import of Json data? Is it supported in the UI?

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,7 +2,42 @@
...
+- Export valid Shapefiles for mappings, including feature types with no geometry
```
Comment (stempler) [inline CHANGELOG.md:19]: Not sure what "Export valid Shapefiles for mappings" means? Is it only about supporting export of Shapefiles w/o geometry?

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,7 +2,42 @@
...
+- Support downloading WFS files to the tmp folder for bucket service
```
Comment (stempler) [inline CHANGELOG.md:20]: I would remove this, it's not a functionality relevant for users.

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,7 +2,42 @@
...
+- Feature: Type selection broken on custom export config
```
Comment (stempler) [inline CHANGELOG.md:21]: I think this should rather be moved to the fixes

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,7 +2,42 @@
...
+- Support adding multiple selections of feature types to be exported to CSV
+- Replace default map with OpenStreetMap
+- Add Boolean and Java Pattern class to Groovy restriction whitelist
+- Update decimal places to 20 for GeoTools
```
Comment (stempler) [inline CHANGELOG.md:25]: From the description it is hard to determine what the change is.
Is this about the changed default decimal places for the Json/GeoJson export?

File: CHANGELOG.md [hunk 9]
```diff
@@ -2,7 +2,42 @@
...
+
+## Maintenance
+- Update Geotools to version 29.1
+- Update dependencies in hale-platform to resolve security issues
+- Fix for the Windows installer
+ 
+### Fixed
+- Fix Excel reader not adding instances for empty rows
```
Comment (stempler) [inline CHANGELOG.md:33]: So now instances are added for empty rows?

File: CHANGELOG.md [hunk 10]
```diff
@@ -2,7 +2,42 @@
...
+- Fix project settings, warnings, include JSON UI extension definition in the build
```
Comment (stempler) [inline CHANGELOG.md:34]: I would remove this as this is not relevant for users

File: CHANGELOG.md [hunk 11]
```diff
@@ -2,7 +2,42 @@
...
+- Fix Shapefile writer to recognize geometries for feature types
```
Comment (stempler) [inline CHANGELOG.md:35]: What does it recognize? The geometry type?

File: CHANGELOG.md [hunk 12]
```diff
@@ -2,7 +2,42 @@
...
+- Fix exporting transformed data leading to a shift of attribute values in Shapefile
```
Comment (stempler) [inline CHANGELOG.md:36]: I think this was a general Shapefile export issue and not limited to transformed data.

File: doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+		<li>Added a parameter to the GeoPackage writer that allows creating tables for all mapping-relevant target types</li>
+	</ul>
+
+	<p>
+		For a full list of changes, please check out the <a
+			href="https://github.com/halestudio/hale/blob/master/CHANGELOG.md#510"
+			target="_blank">changelog on GitHub</a>
+	</p>
+
+	<p>To avoid any compatibility issues when using an existing workspace, we recommend starting with a fresh one when you install
```
Comment (stempler) [inline doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml:22]: Is this still an issue? Are there any instructions on how to do it?

File: doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+<link rel="stylesheet" type="text/css" href="new.css" />
+<title>What's new in hale»studio 5.0.1?</title>
+</head>
+<body>
+	<h1>What's new in hale»studio 5.1.0?</h1>
+
+	<ul>
+		<li>Added an option to overwrite content in Geopackage Writer instead of appending content</li>
+		<li>Added a parameter to the GeoPackage writer that allows creating tables for all mapping-relevant target types</li>
+	</ul>
```
Comment (stempler) [inline doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml:14]: This list should reflect the most important points from the changelog and include the points that are shown in more detail further down.

File: doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+	<p>
+		For a full list of changes, please check out the <a
+			href="https://github.com/halestudio/hale/blob/master/CHANGELOG.md#510"
+			target="_blank">changelog on GitHub</a>
+	</p>
+
+	<p>To avoid any compatibility issues when using an existing workspace, we recommend starting with a fresh one when you install
+		hale»studio 5.1.0</p>
+		
+	<p>The development work for this release was co-funded by the European Health and Digital Executive Agency (HaDEA) under Action No 2018-EU-IA-0093.</p>
```
Comment (stempler) [inline doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml:25]: @thorsten-reitz Is this still up-to-date?

File: CHANGELOG.md (hunk 3 above)
Comment (emanuelaepure10) [reply CHANGELOG.md:16]: Could be enough to keep this 2:
- Implement JSON reader functionality
- Support loading schema from JSON data

File: CHANGELOG.md (hunk 4 above)
Comment (emanuelaepure10) [reply CHANGELOG.md:18]: Any better:

- Added support for importing data from JSON files, enhancing the versatility of data import options.
- Introduced support for loading JSON schema, providing users with greater flexibility in handling JSON data structures.

File: doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml (hunk 15 above)
Comment (thorsten-reitz) [reply doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml:25]: No, as far as I see it would be good to mention Umweltbundesamt Deutschland as wel as the Landesamt für Digitalisierung, Breitband und Vermessung (LDBV) Bayern for their support of the work done on this relase.

Comment (emanuelaepure10) [reply doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml:25]: @thorsten-reitz could this work:

We extend our sincere gratitude to Umweltbundesamt Deutschland and the Landesamt für Digitalisierung, Breitband und Vermessung (LDBV) Bayern for their generous support and collaboration in the development of this release. Their valuable contributions have played a crucial role in enhancing the capabilities and quality of our software.

File: CHANGELOG.md (hunk 9 above)
Comment (emanuelaepure10) [reply CHANGELOG.md:33]: NO instance is added if excel contains an empty row.

File: doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml (hunk 13 above)
Comment (emanuelaepure10) [reply doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml:22]: No idea. Is copy paste from the the other releases. 
I don't see the sense of it. I'll remove it.

File: CHANGELOG.md (hunk 11 above)
Comment (emanuelaepure10) [reply CHANGELOG.md:33]: This is the commit
https://github.com/halestudio/hale/commit/2487be3a3d414b4dea26c5c82b383c8e9ad38943
Doesn't make to much sense to mention it.

File: CHANGELOG.md (hunk 8 above)
Comment (emanuelaepure10) [reply CHANGELOG.md]: The text has been improved:

- Support up to 20 decimal places for source data geometries, ensuring heightened precision and accuracy in the exported data.

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,29 +1,67 @@
...
+- Support for reading multiple sheets from an Excel file simultaneously
+- Enable exporting of multiple feature types to Excel
+- Allow loading Excel with blank rows and sheets for schema and source data
+- Add predefined and custom formatters for Date cells in Excel import
+- Enable exporting feature types with no data in Excel
+- Align Excel with CSV export wizard
+- Prefer properties with short paths as GeoJSON geometry property
+- Add wizard page to select the read mode for GeoJSON/JSON
+- Added support for importing data from JSON files, enhancing the versatility of data import options.
+- Introduced support for loading JSON schema, providing users with greater flexibility in handling JSON data structures.
```
Comment (Kate-Lyndegaard) [inline CHANGELOG.md:21]: Introduced support for loading JSON schema --> Support for loading JSON schema

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,29 +1,67 @@
...
+
+- Support for reading multiple sheets from an Excel file simultaneously
+- Enable exporting of multiple feature types to Excel
+- Allow loading Excel with blank rows and sheets for schema and source data
+- Add predefined and custom formatters for Date cells in Excel import
+- Enable exporting feature types with no data in Excel
+- Align Excel with CSV export wizard
+- Prefer properties with short paths as GeoJSON geometry property
+- Add wizard page to select the read mode for GeoJSON/JSON
+- Added support for importing data from JSON files, enhancing the versatility of data import options.
```
Comment (Kate-Lyndegaard) [inline CHANGELOG.md:20]: Added support for importing data from JSON files --> Support for importing data from JSON files

Comment (Kate-Lyndegaard) [reply CHANGELOG.md:20]: Allow loading Excel with blank rows and sheets for schema and source data --> Allow loading Excel which contains blank rows and sheets for schema and source data. Instances are not added for blank entries.

File: CHANGELOG.md (hunk 12 above)
Comment (Kate-Lyndegaard) [reply CHANGELOG.md:37]: Fix exporting transformed data leading to a shift of attribute values in Shapefile --> Fix shift of attribute values for empty cells in Shapefile export.

File: doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+<link rel="stylesheet" type="text/css" href="new.css" />
+<title>What's new in hale»studio 5.0.1?</title>
+</head>
+<body>
+	<h1>What's new in hale»studio 5.1.0?</h1>
+
+	<ul>
+		<li>update Map using OpenStreetMap</li>
```
Comment (Kate-Lyndegaard) [inline doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml:12]: update Map --> Update Map

File: doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+		<li>Consistent Experience for Excel and CSV</li>
+		<li>Better Excel Support</li>
+		<li>Enhanced Date Formatting for Excel/CSV</li>
```
Comment (Kate-Lyndegaard) [inline doc/plugins/eu.esdihumboldt.hale.doc.user/html/new/5_1_0.xhtml:15]: Excel/CSV --> Excel and CSV

**Review by Kate-Lyndegaard — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 16 above)
Comment (stempler) [reply CHANGELOG.md:21]: We do not support loading [JSON Schema](https://json-schema.org/) as such, so generally I think this phrasing is misleading and may confuse users.

The term before ("Support loading schema from JSON data") was better. Could also be "Support extracting schema from JSON data"

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,29 +1,67 @@
...
+- Introduced support for loading JSON schema, providing users with greater flexibility in handling JSON data structures.
+- Improve the export to Shapefiles including feature types with no geometry
+- Support adding multiple selections of feature types to be exported to CSV
+- Replace default map with OpenStreetMap
+- Add Boolean and Java Pattern class to Groovy restriction whitelist
+- Support up to 20 decimal places for source data geometries, ensuring heightened precision and accuracy in the exported data.
```
Comment (stempler) [inline CHANGELOG.md:26]: Again the question what change this is related to? I assume it is only related to Json? And as a change only related to the Json/GeoJson export, where before a limit of 7(?) was used.

I would maybe rephrase it to

"When exporting data as Json or GeoJson, now up to 20 decimal places are retained in Geometry coordinates"

File: CHANGELOG.md (hunk 9 above)
Comment (stempler) [reply CHANGELOG.md:36]: To me fixing "not adding" sounds like now they are added. Maybe something like this would be more clear:

"Don't create instances from empty rows"

File: CHANGELOG.md (hunk 1 above)
Comment (stempler) [reply CHANGELOG.md:8]: FYI Currently the links are defined [at the end of the document](https://github.com/halestudio/hale/blob/bb04e1dd0e27209353024536837b47e1c3987e0b/CHANGELOG.md?plain=1#L452).

File: CHANGELOG.md (hunk 16 above)
Comment (Kate-Lyndegaard) [reply CHANGELOG.md:21]: @stempler Thank you for providing the clarified description. I just wanted to draw attention to the use of "Introduced". Past tense is not used in any of the other descriptions- it would be good to align it throughout the document.

File: CHANGELOG.md (hunk 1 above)
Comment (emanuelaepure10) [reply CHANGELOG.md:8]: Thank you. I'll add also the missing versions.

**Comment by emanuelaepure10** [comment]
@Kate-Lyndegaard @stempler would you have anything else to add here? thank you

**Review by Kate-Lyndegaard — APPROVED** [review]

File: CHANGELOG.md [hunk 21]
```diff
@@ -362,19 +402,19 @@ on the right-hand rule when exporting data in Json format
 
 ### Changed
 
-- GML encoding: Automatic conversion of polygon geometries to line geometries when there are no possibilities to encode a surface has been changed to produce a *LineString* for a *Polygon* and a *MultiLineString* for a *MultiPolygon*
```
Comment (stempler) [inline CHANGELOG.md:365]: Just wondering - any specific reasons for these changes?

Comment (emanuelaepure10) [reply CHANGELOG.md:365]: No specific reason. And moreover no idea how it happened. 
I have change them back.

**Comment by stempler** [comment]
@emanuelaepure10 Will you create a separate PR adding information in the changes that were done since this PR was created?

AFAIK that would be at least:

- Fix an issue that prevented the more performant index based Join and Merge implementations from being used
- Fix for index based Merge for cases where no Merge key is used
- Update offline versions of INSPIRE schemas to 2024.1.2
- Extend Join functions with optional behavior as "inner joins" where a result is only created if all join conditions are met

**Comment by emanuelaepure10** [comment]
@stempler I've created a new pull request (PR), as I was uncertain whether it would be considered good practice to update an already approved PR.
Thank you
